File: GBFSCron/src/main/java/com/david/GBFSCron/utils/JSONParser.java
package com.david.GBFSCron.utils;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.google.gson.JsonObject;
import com.google.gson.JsonParser;
import org.json.JSONObject;

public class JSONParser {

    public static <T> JSONObject toJSON(Class<T> clazz, Object object){
        Gson gson = new GsonBuilder().create();
        return new JSONObject(gson.toJson((T) object));
    }

    public static <T> T parseJSON(String json, Class<T> clazz) {
        JsonParser parser = new JsonParser();
        JsonObject jsonObject = parser.parse(json).getAsJsonObject();
        Gson gson = new GsonBuilder().create();

        return gson.fromJson(jsonObject, clazz);
    }
}

File: GenerateGBFSDataset/src/main/scala/com/david/GenerateGBFS/utils/JSONParser.scala
package com.david.GenerateGBFS.utils

import com.david.GenerateGBFS.json.JSON
import com.david.GenerateGBFS.model.MODEL
import org.json4s._
import org.json4s.jackson.JsonMethods._
object JSONParser {
  implicit val formats = DefaultFormats

  def parseJSON[T <: JSON[_ <: MODEL] : Manifest](json: String, clazz: Class[_ <: MODEL]): JSON[_ <: MODEL] = { // Use Manifest if you want to pass the object directly
    return parse(json).extract[T]
  }
}
